models/bpf.go

```diff
@@ -0,0 +1,132 @@
+// Copyright Contributors to the L3AF Project.
+// SPDX-License-Identifier: Apache-2.0
+
+package models
+
+import (
+	"github.com/cilium/ebpf"
+	"github.com/l3af-project/l3afd/config"
+)
+
+// BPF defines run time details for BPFProgram.
+type BPF interface {
+	// Name returns the BPFProgram Name
+	Name() string
+
+	// Artifact returns the BPFProgram Artifact
+	Artifact() string
+
+	// MapName returns the map name of the BPFProgram
+	MapName() string
+
+	// ProgId returns the program ID of the BPFProgram
+	ProgId() int
+
+	// SeqId return the sequence ID of the BPFProgram
+	SeqId() int
+
+	// AdminStatus return the admin status of the BPFProgram
+	AdminStatus() string
+
+	// Version returns the version of the BPFProgram
+	Version() string
+
+	// RestartCount returns the restart count
+	GetRestartCount() int
+
+	// GetProgram return the underlying BPFProgram
+	GetProgram() *BPFProgram
+
+	// GetMapNamePath returns the MapNamePath
+	GetMapNamePath() string
+
+	// GetPrevMapNamePath returns the PrevMapNamePath
+	GetPrevMapNamePath() string
+
+	// AddToRestartCount increases the RestartCount by the passed value
+	AddToRestartCount(value int)
+
+	// UpdateAdminStatus updates the admin status of the BPFProgram to rhe passed string
+	UpdateAdminStatus(value string)
+
+	// UpdatesProgram updates the underlying BPFProgram to the passed BPFProgram struct
+	UpdateProgram(value *BPFProgram)
+
+	// UpdatePrevMapNamePath updates the PrevMapNamePath to the passed string.
+	UpdatePrevMapNamePath(value string)
+
+	// Stop returns the last error seen, but stops bpf program.
+	// Clean up all map handles.
+	// Verify next program pinned map file is removed
+	Stop(ifaceName, direction string, chain bool) error
+
+	// Start returns the last error seen, but starts bpf program.
+	// Here initially prevprogmap entry is removed and passed to the bpf program
+	// After starting the user program, will update the kernel progam fd into prevprogram map.
+	// This method waits till prog Namefd entry is updated, else returns error assuming kernel program is not loaded.
+	// It also verifies the next program pinned map is created or not.
+	Start(ifaceName, direction string, chain bool) error
+
+	// Update updates the config map_args
+	Update(ifaceName, direction string) error
+
+	// VerifyAndGetArtifacts checks if binary already exists and downloads artifacts from the specified eBPF repo
+	VerifyAndGetArtifacts(conf *config.Config) error
+
+	// IsRunning checks the status of the user program and returns if it is running or not
+	IsRunning() (bool, error)
+
+	// GetBPFMap returns a BPFMap containg the pinned map in case of a TC program and a new map for a given ID in case of XDP program
```

Comment:
```suggestion
	// GetBPFMap returns a BPFMap containing the pinned map in case of a TC program and a new map for a given ID in case of XDP program
```

I think, however, it's not good to have behavior that varies by program type as it makes it hard to extend this to more program types.  Instead I think there should either be separate APIs or an argument to the API to determine whether to get a pinned map vs a new map.